File: src/internal/schema/experiment.go
package sc

import "time"

type CreateExperimentRequest struct {
	ExperimentName        string `json:"name"`
	ExperimentDescription string `json:"description"`
}

type CreateExperimentResponse struct {
	ExperimentID uint64 `json:"id"`
}

type ExperimentSummary struct {
	ExperimentID          uint64    `json:"id"`
	ExperimentName        string    `json:"name"`
	ExperimentDescription string    `json:"description"`
	CreatedAt             time.Time `json:"createdAt"`
	UpdatedAt             time.Time `json:"updatedAt"`
}

type ExperimentDetails struct {
	ExperimentSummary ExperimentSummary `json:"experimentSummary"`
	RunDetails        []RunDetails      `json:"runDetails"`
}

type GetExperimentResponse struct {
	ExperimentDetails *ExperimentDetails `json:"experimentDetails"`
}

type GetExperimentSummariesResponse struct {
	ExperimentSummaries []*ExperimentSummary `json:"experimentSummaries"`
}

File: web/src/lib/types/base/experiment.ts
import type { RunDetails } from './run';

export type ExperimentSummary = {
    id: number;
    name: string;
    description: string;
    createdAt: Date;
    updatedAt: Date;
};

export type ExperimentDetails = {
    experimentSummary: ExperimentSummary;
    runDetails: RunDetails[];
};
